plugins/org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/hierarchy/DefaultHierarchyNode.xtend

```diff
@@ -27,12 +27,12 @@ class DefaultHierarchyNode implements HierarchyNode {
 	IEObjectDescription element
 
 	@Accessors(PUBLIC_GETTER)
-	val locations = <HierarchyNodeLocation>newArrayList
+	val references = <HierarchyNodeReference>newArrayList
```

Comment:
The getter should not expose the kind of list implementation, so better add an explicit field type `List <HierarchyNodeReference>`